File: solutions/python/matrix/2/matrix.py
"""Matrix exercise"""
class Matrix:
    def __init__(self, matrix_string):
        self.rows = [
            [
                int(string) for string in row.split(' ')
            ] for row in matrix_string.split("\n")
        ]

        self.columns = [[] for i in range(len(self.rows[0]))]

        for i in range(len(self.rows)):
            for j in range(len(self.rows[0])):
                self.columns[j].append(self.rows[i][j])

    def row(self, index):
        print(self.rows)
        return self.rows[index - 1]

    def column(self, index):
        print(self.columns)
        return self.columns[index - 1]

File: solutions/ruby/matrix/1/matrix.rb
# Write your code for the 'Matrix' exercise in this file. Make the tests in
# `matrix_test.rb` pass.
#
# To get started with TDD, see the `README.md` file in your
# `ruby/matrix` directory.

class Matrix
  attr_reader :rows, :columns

  def initialize(string)
    @rows = string.split("\n").map { |row| row.split(' ').map(&:to_i) }
    @columns = Array.new(@rows[0].length) { [] }

    @rows.each_index do |i|
      @rows[i].each_index do |j|
        @columns[j][i] = rows[i][j]
      end
    end
  end
end
